hdfscontents/hdfsmanager.py

```diff
@@ -375,16 +380,37 @@ def rename_file(self, old_path, new_path):
 
         new_hdfs_path = to_os_path(new_path, self.root_dir)
         old_hdfs_path = to_os_path(old_path, self.root_dir)
-
+        jupyter_configuration = self.get_notebook_jupyter_configuration_xatrr(old_hdfs_path)
         # Should we proceed with the move?
         if self._hdfs_exists(new_hdfs_path):
             raise web.HTTPError(409, u'File already exists: %s' % new_path)
 
         # Move the file
         try:
             self._hdfs_move_file(old_hdfs_path, new_hdfs_path)
+            self.set_notebook_jupyter_configuration_xatrr(new_hdfs_path, jupyter_configuration)
         except Exception as e:
             raise web.HTTPError(500, u'Unknown error renaming file: %s %s' % (old_path, e))
 
     def info_string(self):
         return "Serving notebooks from HDFS directory: %s" % self.root_dir
+
+    def get_notebook_jupyter_configuration_xatrr(self, hdfs_path):
+        self.log.debug("Getting jupyter configuration xattr for %s", hdfs_path)
+        jupyter_configuration = {}
+        if ".ipynb" not in hdfs_path:
+            return jupyter_configuration
+        try:
+            jupyter_configuration = xattr.get_xattr(hdfs_path, self.jupyter_configuration_xattr_name)
+        except Exception as e:
+            self.log.debug(u'Failed to get jupyter configuration xattr for %s %s', hdfs_path, e, exc_info=True)
+        return jupyter_configuration
+
+    def set_notebook_jupyter_configuration_xatrr(self, hdfs_path, jupyter_configuration):
```

Comment:
Also check that xattr is loaded here too.

Reply:
When xattr is not laoded `get_notebook_jupyter_configuration_xatrr` will return an empty set. So there will be no problem here.